megamek/src/megamek/client/bot/princess/PathRanker.java

```diff
@@ -49,7 +49,7 @@ public PathRanker(Princess princess) {
         owner = princess;
     }
 
-    private Map<MovePath.Key, Double> pathSuccessProbabilities = new HashMap<MovePath.Key, Double>();
+    private Map<MovePath.Key, Double> pathSuccessProbabilities = new HashMap<>();
```

Comment:
This change doesn't seem correct. You've removed the strict typing in half of the statement for no reason that I can see...

Reply:
It's standard Java 7+ convention to type inference in place of parameterized types:
https://docs.oracle.com/javase/tutorial/java/generics/types.html#diamond

Reply:
I had forgotten about that change, and was having nightmares about the hundreds of places I've added parameterized types when Eclipse was throwing over 2000 warnings for them a few years ago. I'll approve the PR ;)